pkg/crc/machine/machine.go

```diff
@@ -128,6 +130,15 @@ func Start(startConfig StartConfig) (StartResult, error) {
 			return *result, errors.Newf("Error getting the state for host: %v", err)
 		}
 
+		if err := ssh.GenerateSSHKey(constants.GetPrivateKeyPath()); err != nil {
```

Comment:
I'd move this to `updatePublicSSHKey` (and rename it to something different as the 'Public' part of the name would be odd), `Start` is too long as it is, so I'd prefer that we only call helpers from it.